### ISSUE TITLE
Youtube music, can play on browser, but cannot download

### ISSUE THREAD

**Issue by zugnujaspu** [yt-dlp/yt-dlp#6351]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2023.02.17** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### **_But can play on browser_**

https://music.youtube.com/playlist?list=PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r

> Got the URL from https://www.thefirsttake.jp/#playlist
> ... https://lnk.to/from_tft

Have also tried `--extractor-args "youtube:player-client=..."` with ...

- web_music
- android_music
- web_embedded

All not working

Please help.

Thanks!

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-i', '-f', 'bestaudio[ext=m4a]', '--embed-metadata', '--embed-thumbnail', '--playlist-end', '1', 'https://music.youtube.com/playlist?list=PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r&src=Linkfire&lId=c64f0415-d1b6-40a6-9a92-d925887fabec&cId=d3d58fd7-4c47-11e6-9fd0-066c3e7a8751']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.02.17 [a0a7c01] (linux_exe)
[debug] Python 3.10.9 (CPython x86_64 64bit) - Linux-6.0.0-0.deb11.6-amd64-x86_64-with-glibc2.31 (OpenSSL 3.0.8 7 Feb 2023, glibc 2.31)
[debug] exe versions: ffmpeg 5.0.2 (setts), ffprobe 5.0.2
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.02.17, Current version: 2023.02.17
yt-dlp is up to date (2023.02.17)
[youtube:tab] Extracting URL: https://music.youtube.com/playlist?list=PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r&src=Linkfire&lId=c64f0415-d1b6-40a6-9a92-d925887fabec&cId=d3d58fd7-4c47-11e6-9fd0-066c3e7a8751
[youtube:tab] PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r: Downloading webpage
WARNING: [youtube:tab] YouTube said: INFO - 94 unavailable videos are hidden
[youtube:tab] PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: FROM THE FIRST TAKE
[debug] Skipping writing playlist thumbnail
[youtube:tab] Playlist FROM THE FIRST TAKE: Downloading 1 items of 120
[download] Downloading item 1 of 1
[youtube] Extracting URL: https://music.youtube.com/watch?v=IJzKfLBlFnk
[youtube] IJzKfLBlFnk: Downloading webpage
[youtube] IJzKfLBlFnk: Downloading android player API JSON
[youtube] IJzKfLBlFnk: Downloading android music player API JSON
[youtube] IJzKfLBlFnk: Downloading web music client config
[youtube] IJzKfLBlFnk: Downloading player 9419f2ea
[youtube] IJzKfLBlFnk: Downloading web music player API JSON
ERROR: [youtube] IJzKfLBlFnk: Video unavailable. This video is not available
  File "yt_dlp/extractor/common.py", line 693, in extract
  File "yt_dlp/extractor/youtube.py", line 4034, in _real_extract
  File "yt_dlp/extractor/common.py", line 1171, in raise_no_formats

[download] Finished downloading playlist: FROM THE FIRST TAKE
```

**Comment by coletdjnz** [comment]
The first video in the playlist is unavailable, but these are hidden on YouTube music. We grab from normal YouTube, and if you go on there and enable unavailable videos you will see the first video is unavailable.

https://www.youtube.com/playlist?list=PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r

> Unavailable videos are also listed for YouTube playlists. Use --compat-options no-youtube-unavailable-videos to remove this

**Comment by zugnujaspu** [comment]
But if you go to the URL
>https://music.youtube.com/playlist?list=PLeLvSt3A0DdnTOLYurO9JFuQsNS1Lrq4r

and click the first song, you can play it
>the "Michiz..... something"


So I was wondering why can play but cannot download... strange

**Comment by pukkandan** [comment]
> Unavailable videos are also listed for YouTube playlists. Use --compat-options no-youtube-unavailable-videos to remove this

**Comment by zugnujaspu** [comment]
I meant if can play, then it is NOT unavailable, right?

Also meaning the detection is wrong or something

**Comment by pukkandan** [comment]
hm.. It appears youtube.com and music.youtube provides entirely different items in this playlist. We can't fix it unless https://github.com/yt-dlp/yt-dlp/issues/622

**Comment by Megver83** [comment]
IDK if this workaround will work for all songs, but I tried to download a whole album with this issue and although the YouTube Music URL didn't work, it did with [Beatbump](https://github.com/snuffyDev/Beatbump)

Here are the URLs I tried:
https://piped.kavin.rocks/playlist?list=OLAK5uy_kS8fM7WR8X7IK9_0WrfX68nuS8qfXXNjw (returns "video unavailable")
https://beatbump.ml/release?id=MPREb_TXGFRfQQ7Pd (works)

**Comment by Zezombye** [comment]
There should indeed be some kind of flag to force extraction from music.youtube.com instead of youtube.com (aren't the API formats the same?)

Yt-dlp could even automatically check if the video is unavailable and if it is automatically generated. For example, viewing the source code of https://www.youtube.com/watch?v=jUjso3vrlXI still gives access to the description, and if it ends with `"\n\nAuto-generated by YouTube."` then check https://music.youtube.com/watch?v=jUjso3vrlXI where it gives the new ID: `\"watchEndpoint\":{\"videoId\":\"Ujwvi1RGcKw\"`

**Comment by adyanth** [comment]
There are some useful behaviors noted in this issue fyi: https://github.com/yt-dlp/yt-dlp/issues/13376#issuecomment-2931932412

TLDR; looks like the videoIds accessible are different based on the country and the logged in status. For most of them, YouTube redirects the request to the correct videoId that is available to the region + account (YTPremium impacts this maybe?). But this redirect is not handled by yt-dlp, since I believe it looks up metadata on YouTube which isn't accessible.

So looks like https://github.com/yt-dlp/yt-dlp/issues/622 is needed for this after all.
